### PR TITLE
feat: Check if `./target` is writable

### LINKED ISSUE #83: Ensure that ./target is writable
When the plugin starts it should:
- create `./target` if it does not exist.
- check that `./target` is writable if it exists.

If either operation fails, it should NOT start the Flix compiler, but instead report an error suggesting to the user that he should open the flix file in project in a writable dir.

### PR DESCRIPTION
**Pull request by plougsgaard**
Fixes #83

### CODE REVIEW

File: client/src/util/ensureTargetWritable.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ */
+
+import * as path from 'path'
+import * as fs from 'fs'
+
+export default function ensureTargetWritable (workspaceFolder: string): boolean {
+  const targetPath = path.join(workspaceFolder, 'target')
+  if (!fs.existsSync(targetPath)) {
+    fs.mkdirSync(targetPath)
```
Comment (magnus-madsen) [inline client/src/util/ensureTargetWritable.ts:23]: This is not inside the try catch, so could it not throw an exception? I dont know the node api that well..

Comment (plougsgaard) [reply client/src/util/ensureTargetWritable.ts:23]: You are correct.

https://nodejs.org/api/errors.html
> With few exceptions, Synchronous APIs (any blocking method that does not accept a callback function, such as fs.readFileSync), will use throw to report errors.

Comment (magnus-madsen) [reply client/src/util/ensureTargetWritable.ts:23]: Technically correct?

Comment (plougsgaard) [reply client/src/util/ensureTargetWritable.ts]: Yes, I've updated the PR. Forgot to push before let's say.

File: client/src/extension.ts [hunk 2]
```diff
@@ -174,6 +175,11 @@ export async function activate (context: vscode.ExtensionContext, launchOptions:
   const workspaceFolders = _.map(_.flow(_.get('uri'), _.get('fsPath')), vscode.workspace.workspaceFolders)
   const workspaceFiles: [string] = _.map(vsCodeUriToUriString, (await vscode.workspace.findFiles(FLIX_GLOB_PATTERN)))
 
+  // Make sure we can write to `./target`
+  if (!ensureTargetWritable(_.first(workspaceFolders))) {
+    throw new Error('Cannot write to "target" folder.')
```
Comment (magnus-madsen) [inline client/src/extension.ts:180]: should this be a notification or does an exception automatically become that?

Comment (plougsgaard) [reply client/src/extension.ts:180]: Uncaught exceptions are automatically picked up by VSCode. It then terminates the plugin showing the error message specified.

Comment (magnus-madsen) [reply client/src/extension.ts:180]: Great! So this is ready for merge?

Comment (plougsgaard) [reply client/src/extension.ts:180]: Yes!

**Pull request merged**